strmangle.go

```diff
@@ -17,8 +17,8 @@ var (
 	idAlphabet    = []byte("abcdefghijklmnopqrstuvwxyz")
 	smartQuoteRgx = regexp.MustCompile(`^(?i)"?[a-z_][_a-z0-9\-]*"?(\."?[_a-z][_a-z0-9]*"?)*(\.\*)?$`)
 
-	rgxEnum            = regexp.MustCompile(`^enum(\.[a-z0-9_]+)?\((,?'[^']+')+\)$`)
-	rgxEnumIsOK        = regexp.MustCompile(`^(?i)[a-z][a-z0-9_\s]*$`)
+	rgxEnum            = regexp.MustCompile(`^enum(\.[a-zA-Z0-9_]+)?\((,?'[^']+')+\)$`)
+	rgxEnumIsOK        = regexp.MustCompile(`^(?i)[a-zA-Z][a-zA-Z0-9_\s]*$`)
 	rgxEnumShouldTitle = regexp.MustCompile(`^[a-z][a-z0-9_]*$`)
```

Comment:
I think we should also edit this for the 2nd part of the regex so that both `myCustomType` and `my_custom_type` are title cased. Also can you add tests to check this? I'm just eyeballing it here.

```suggestion
	rgxEnumShouldTitle = regexp.MustCompile(`^[a-z][a-zA-Z0-9_]*$`)
```

Reply:
Also done. And Tests added.